packages/inspector/src/components/EntityInspector/LightSourceInspector/utils.ts

```diff
@@ -0,0 +1,76 @@
+import type { PBLightSource } from '@dcl/ecs';
+import { toColor3, toHex } from '../../ui/ColorField/utils';
+import type { LightInput } from './types';
+import { LightKind } from './types';
+
+export const isValidInput = (input: LightInput) => {
+  const intensity = Number(input.intensity);
+  if (isNaN(intensity)) return false;
+  if (intensity < 0) return false;
+  if (input.type === LightKind.SPOT) {
+    const ia = Number(input.spot?.innerAngle ?? 0);
+    const oa = Number(input.spot?.outerAngle ?? 0);
+    if (isNaN(ia) || isNaN(oa)) return false;
+    if (ia < 0 || ia > 180 || oa < 0 || oa > 180) return false;
+  }
+  return true;
+};
+
+export const fromComponent = (value: PBLightSource): LightInput => {
+  const base: LightInput = {
+    type: value.type?.$case === 'spot' ? LightKind.SPOT : LightKind.POINT,
+    active: value.active === undefined ? true : !!value.active,
+    color: toHex(value.color).toUpperCase(),
+    intensity: String(value.intensity ?? 16000),
+    shadow: !!value.shadow,
+    shadowMaskSrc:
+      value.shadowMaskTexture?.tex?.$case === 'texture'
+        ? value.shadowMaskTexture.tex.texture.src
+        : '',
+    range: String(value.range ?? -1),
+  };
+
+  if (value.type?.$case === 'spot') {
+    base.spot = {
+      innerAngle: String(value.type.spot.innerAngle ?? 30),
+      outerAngle: String(value.type.spot.outerAngle ?? 40),
+    };
+  }
+
+  return base;
+};
+
+export const toComponent = (input: LightInput): PBLightSource => {
+  const isSpot = input.type === LightKind.SPOT;
+  const type = isSpot
+    ? {
+        $case: 'spot',
+        spot: {
+          innerAngle: Number(input.spot?.innerAngle || 30),
+          outerAngle: Number(input.spot?.outerAngle || 40),
```

Comment:
in case the `inner|outer` angle is `0`, should we use the default value?

Reply:
no, I think that would be very unexpected

if you make either of the two 0 you won't see any light, but at least it's expected
and maybe you want to make a transition where for example the aperture starts at 0 and then opens up